crates/rattler_cache/src/package_cache/cache_lock.rs

```diff
@@ -196,7 +221,14 @@ impl CacheRwLock {
 }
 
 impl CacheRwLock {
+    /// Reads the revision from the cache lock file.
     pub fn read_revision(&mut self) -> Result<u64, PackageCacheError> {
+        self.file.lock().rewind().map_err(|e| {
```

Comment:
Would be good to lock once so that another thread doesnt move the position in the file.

Reply:
Good point!